src/envs/touch_point.py

```diff
@@ -177,3 +178,22 @@ def _Touched_holds(self, state: State, objects: Sequence[Object]) -> bool:
         ty = state.get(target, "y")
         dist = np.sqrt((rx - tx)**2 + (ry - ty)**2)
         return dist < self.action_magnitude * self.touch_multiplier
+
+    def get_event_to_action_fn(
+            self) -> Callable[[State, matplotlib.backend_bases.Event], Action]:
+        logging.info("Controls: mouse click to move")
+
+        def _event_to_action(state: State,
+                             event: matplotlib.backend_bases.Event) -> Action:
+            assert event.key is None, "Keyboard controls not allowed."
```

Comment:
does this crash everything? or just wait for another action from the user? the latter is preferred

Reply:
it crashes everything, it's not super easy to wait for another action from the user because we're calling `plt.waitforbuttonpress()` which hangs until either a mouse click or a keystroke, but then we'd need to detect which one actually happened and receive information from the env about which of them is "legal". let's just leave it as is for now, we can address this later if we really care

Reply:
btw for similar reasons, it will also crash everything (with an informative error message) if you accidentally click outside the bounds of the grid (but still within the figure window). try not to do that on task 999! 😛

Reply:
You joke, but that would actually be tragic lol. Is there any way we can save as we go?

Reply:
we can make that a standalone PR that saves each task separately during data collection and then deletes all of them when it reaches the end of data collection. thoughts? if that sounds good, i can make an issue

Reply:
(this can happen regardless of the setting of `demonstrator`)

Reply:
something like that sounds good, +1 to making it a separate issue and PR

Reply:
hmm we would also need it to load from the previously saved tasks if there was partial progress

Reply:
regardless, i'm merging this because whatever we choose to do can be a standalone PR